modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLSSOForAdminLoginTestCase.java

```diff
@@ -0,0 +1,183 @@
+    public void testSSOForAdminLogin() {
+
+        try {
+            // Get to management console login page
+            HttpResponse response = Utils.sendGetRequest(MANAGEMENT_CONSOLE_LOGIN_URL, USER_AGENT, httpClient);
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), 200,
+                    "User may have already logged in");
+
+            // Extract saml request from the response and send the post request to samlsso endpoint
+            String samlRequest = Utils.extractDataFromResponseForManagementConsoleRequests(response, "value", 1);
+            response = sendSAMLMessage(SAML_SSO_URL, CommonConstants.SAML_REQUEST_PARAM, samlRequest, config);
+            EntityUtils.consume(response.getEntity());
+            response = Utils.sendRedirectRequest(response, USER_AGENT, ACS_URL, SAML_SSO_URL, httpClient);
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), 200,
+                    "SAML SSO Login failed for " + config);
+
+            // check whether the user is logged in
+            response = Utils.sendGetRequest(MANAGEMENT_CONSOLE_HOME_URL, USER_AGENT, httpClient);
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), 200,
+                    "SAML SSO Login failed for " + config);
+        } catch (Exception e) {
+            Assert.fail("SAML SSO Login test failed for " + config, e);
+        }
+    }
+
+    @DataProvider(name = "samlConfigProvider")
+    public static SAMLConfig[][] samlConfigProvider() {
+
+        return new SAMLConfig[][]{
+                {new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, User.SUPER_TENANT_ADMIN, HttpBinding.HTTP_REDIRECT,
+                        ClaimType.LOCAL, App.MANAGEMENT_CONSOLE_SSO_APP_WITH_SIGNING)}
+        };
+    }
+
+    /**
+     * This method will reset the deployment.toml file to default configurations.
+     */
+    private void changeISConfiguration() throws AutomationUtilException, IOException, XPathExpressionException {
+
+        serverConfigurationManager = new ServerConfigurationManager(isServer);
+
+        log.info("Changing deployment.toml file configs to enable sso for admin login.");
+        String carbonHome = Utils.getResidentCarbonHome();
+        File defaultConfigFile = getDeploymentTomlFile(carbonHome);
+        File configuredIdentityXML = new File(getISResourceLocation() + File.separator + "saml"
+                + File.separator + "saml-sso-for-admin-console.toml");
+        serverConfigurationManager.applyConfigurationWithoutRestart(configuredIdentityXML, defaultConfigFile, true);
+
+        // creating a new sp from file based configs
+        File SAMLSSOSpXml = new File(FrameworkPathUtil.getSystemResourceLocation() + "artifacts" + File.separator + "IS"
+                + File.separator + "saml" + File.separator + "filebasedspidpconfigs" + File.separator +
+                "management-console-sso-carbonServer.xml");
```

Comment:
shall we add a separate method to build paths (if it already doesn't exist in a util class) and reuse? Would improve readability

Reply:
Resolved with 1480ccb9e6cfa7381bd12700f950cda6ebcda7f5